hangupsbot/plugins/tldr.py

```diff
@@ -35,9 +35,16 @@ def tldrecho(bot, event, *args):
     else:
         # No path was found. Is this your first setup?
         new_tldr = 0
+    
+    if tldr_echo_options[new_tldr] is not "GLOBAL":
+        # Update the tldr_echo setting
+        bot.memory.set_by_path(['conversations', event.conv_id, 'tldr_echo'], new_tldr)
+    else:
+        # If setting is global then clear the conversation memory entry
+        conv_settings = bot.memory.get_by_path(['conversations', event.conv_id])
+        del conv_settings['tldr_echo'] # remove setting
+        bot.memory.set_by_path(['conversations', event.conv_id], conv_settings)
```

Comment:
Re-reading this, I should point out that the previous code had no mechanism for the second of the two pseudocode examples above as the method was to remove the memory entry and revert to using config default.